tests/admin/test-class-boldgrid-backup-admin-auto-updates.php

```diff
@@ -0,0 +1,368 @@
+	 */
+	private $auto_updates;
+
+	/**
+	 * Default Test Settings.
+	 *
+	 * @since SINCEVERSION
+	 * @var array
+	 */
+	private $default_test_settings;
+
+	/**
+	 * Install Plugin.
+	 *
+	 * @since SINCEVERSION
+	 * @param string $slug Plugin Slug.
+	 * @param string $version Plugin Version to Install.
+	 */
+	public function install_plugin( $slug, $version ) {
+		global $wp_filesystem;
+		include_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+		include_once ABSPATH . '/wp-admin/includes/file.php';
+		WP_Filesystem();
+
+		$plugin_info     = plugins_api(
+			'plugin_information',
+			array(
+				'slug'   => $slug,
+				'fields' => array(
+					'downloadlink' => true,
+					'versions'     => true,
+				),
+			)
+		);
+		$this_plugin_dir = ABSPATH . 'wp-content/plugins/' . $slug;
+
+		$url      = isset( $plugin_info->versions[ $version ] ) ? $plugin_info->versions[ $version ] : '';
+		$zip_file = ! empty( $url ) ? download_url( $url ) : '';
+		if ( true === is_dir( ABSPATH . 'wp-content/plugins/' . $slug ) ) {
+			$deleted = $wp_filesystem->delete( $this_plugin_dir, true );
+		}
+		$unzip = ! empty( $zip_file ) ? unzip_file( $zip_file, ABSPATH . 'wp-content/plugins/' ) : false;
+		return $unzip;
+	}
+
+	/**
+	 * Setup Tests.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function setUp() {
+		$this->default_test_settings = array(
+			'auto_update' => array(
+				'timely-updates-enabled' => '0',
+				'days'                   => 0,
+				'wpcore'                 => array(
+					'all'         => '0',
+					'major'       => '0',
+					'minor'       => '1',
+					'dev'         => '0',
+					'translation' => '0',
+				),
+				'plugins'                => array(
+					'default' => '0',
+				),
+				'themes'                 => array(
+					'default' => '0',
+				),
+			),
+		);
+	}
+
+	/**
+	 * Test Constructor.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function test_constructor() {
+		// Test that WP's Auto Updater is NOT disabled after init.
+		$auto_updates = new Boldgrid_Backup_Admin_Auto_Updates();
+		$this->assertFalse( apply_filters( 'automatic_updater_disabled', true ) );
+	}
+
+	/**
+	 * Test is_premium_done.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function test_is_premium_done() {
+		// Verify that is_premium_done returns false if premium plugin is not activated.
+		$auto_updates = new Boldgrid_Backup_Admin_Auto_Updates();
+		$this->assertFalse( $auto_updates->is_premium_done() );
+
+		// Verify that it returns true if it is activated.
+		update_option( 'active_plugins', array( 'boldgrid-backup-premium/boldgrid-backup-premium.php' ) );
+		$this->assertTrue( $auto_updates->is_premium_done() );
+	}
+
+	/** Test set_settings.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function test_set_settings() {
+		// Test that an unset 'boldgrid_backup_settings' option returns an empty array.
+		$auto_updates = new Boldgrid_Backup_Admin_Auto_Updates();
+		$this->assertEquals( array(), $auto_updates->settings );
```

Comment:
I forgot to point this out earlier, but use `[]` instead of `array()`.

Reply:
Newest version of Wordpress Coding standards require array() instead of [] . I understand that the use of array() and [] seems to be interchangeable in some places, but I thought it would be a good idea to use array() instead, since that is what Wordpress coding standards recommends. 

[https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#declaring-arrays](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#declaring-arrays)

Reply:
@cssjoe you're not going to be happy about this.

Reply:
@cssjoe and I had that talk already. lol